### ISSUE TITLE
How can the model differentiate Ground、Sky、Plants、Non-stuff after training？

### ISSUE THREAD

**Issue by ShiAngWang** [Max-Manning/autoregunsupseg#2]
I'm very curious about that how the model have the ablity to differentiate Ground、Sky、Plants、Non-stuff without explicit labels but with just no-label training? How does the model know to partition these kinds of parts rather than such as human-animal-plant partition? How does the model do the segment-label map(This question, in my mind, maybe implemented by the  Hungarian algorithm)?

Looking forward to your reply!!

**Comment by Max-Manning** [comment]
Hi - relating to your first question, the output labels are always arbitrary since it is an unsupervised algorithm. It is related to other "contrastive learning" techniques which try to find natural cluseters in the training data by considering their visual similarity. As you pointed out, it does not know the the semantic classes of these clusters because there are no training labels. This is why the hungarian algorithm is needed. It associates each of the arbitrary output classes with a semantic class by computing the best one-to-one assignment between them.

I can't really comment on why your results look the way they do, except to mention that coco-stuff 15 is a very challenging dataset for unsupervised deep learning models.  I was never able to recreate the accuracy that the original authors claimed in their paper, and the poor preservation of object boundaries that you observe in your results is pretty typical for this technique. It is more effective on datasets where each class is more consistent in appearance (for example the potsdam dataset).

**Comment by ShiAngWang** [comment]
Thank you for your reply very much!
So this code repo is not the original code repo that the paper author created?
And I wanna ask that have you output  result images in CocoStuff3 or CocoStuff15 just like the perfect result images shown in the paper? If you have, can you please show me some results?
The results now I achieved is really not very satisfying.

**Comment by Max-Manning** [comment]
No, I am not affiliated in any way with the original authors. I did my best to recreate the method with the information that was included in the paper, but obviously there are some differences from their code (which may explain the poor performance on the CocoStuff Dataset.) I should also point out that the network architectures in this repo are smaller than the ones specified in the original paper to accomodate for GPU memory constraints (they use 5 residual blocks, I only used 4.)

Anyway, I lost interest in working on this method a while ago so I won't be able to help much beyond that. If you do some experiments and discover what I've done wrong, I'd be grateful to hear about it!
